feat(skeleton): refactor skeleton loaders to demonstrate patterns

wbarbee opened the pull request:
## Summary

Refactor skeleton animation and styling and apply them in clearly demonstrable patterns for:
1) Card
2) Data Table

## ADO Story or GitHub Issue Link

[User Story](https://dev.azure.com/Kyndryl/Shidoka%20-%20Bridge%20Design%20System/_workitems/edit/1991926)

## Figma Link

Link here (if applicable).

## Checklist

- [ ] Used Conventional Commit messages as outlined in the contributing guide.
  - [ ] Noted breaking changes (if any).
- [ ] Documented/updated all props, events, slots, parts, etc with JSDoc.
  - [ ] Ran the `analyze` command to update Storybook docs.
- [ ] Added/updated Stories with controls to cover all variants.
- [ ] Ran `test` locally to address any failures.
- [ ] Added/updated the Figma link for the Story's Design tab.
- [ ] Added any new component exports to the src/index.ts file

## Testing Instructions

Provide guidance to reviewers/testers here.

## Screenshots

(if any)

kumarivijeta commented on src/components/reusable/table/table.skeleton.ts:
```diff
@@ -0,0 +1,273 @@
+                    </kyn-tr>
+                  `
+                )}
+            </kyn-tbody>
```
@wbarbee Good to have pagination skeleton as well ?

wbarbee commented:
@kumarivijeta -- i believe we've addressed all of the components that we can for this round. we'll leave Charts and Accordion for a later date (i updated the ADO to reflect this).

today i added a modal skeleton for good measure, though i know this component is less likely to receive data from an API.

otherwise just made some structural changes to hopefully improve things. whenever you get a chance, please take one last pass at each component and change/revert anything that you see that would improve things.

then we can mark it ready for review. thank you!

kumarivijeta reviewed (commented):
@wbarbee Skeleton's looks good to me.

brian-patrick-3 reviewed (commented):
A few thoughts:

1. Modal and Side Drawer: I'm not quite sure having skeletons of these components makes sense. I can picture a scenario where an application has to wait for the content within to load, but not one where the modal itself has to load.
2. Tabs: Maybe the same comment as above.
3. Card Pattern Skeletons: I'm wondering if we should provide actual components instead of sample components here. It would make sense, especially once we move over the Card component from Foundation.

wbarbee commented on src/components/reusable/loaders/skeleton.ts:
```diff
@@ -1,28 +1,85 @@
+   * Optional: Predefined skeleton types with default styling
+   */
+  @property({ type: String, reflect: true })
+  elementType?:
```
makes sense, we can do that. the width and height properties make these fully customizable to the dev, but i thought that setting up some specific elements would be good. more generic is probably better

wbarbee commented on src/components/reusable/loaders/skeleton.scss:
```diff
@@ -1,35 +1,189 @@
 }
 
-:host([inline]) {
+:host(.inline) {
```
this was not intended, will fix

wbarbee commented:
> A few thoughts:
> 
> 1. Modal and Side Drawer: I'm not quite sure having skeletons of these components makes sense. I can picture a scenario where an application has to wait for the content within to load, but not one where the modal itself has to load.
> 2. Tabs: Maybe the same comment as above.
> 3. Card Pattern Skeletons: I'm wondering if we should provide actual components instead of sample components here. It would make sense, especially once we move over the Card component from Foundation.

1 + 2. we just went through and made component files for whatever could have some conceivable need for skeletons. easy to kill these.
2. for this one, are you suggesting that we remove these patterns and revisit cards and accordion when those components are migrated over from Foundation?

kumarivijeta commented on src/components/reusable/tag/tagGroup.stories.js:
```diff
@@ -54,3 +55,30 @@ export const TagGroup = {
 };
+
+export const Skeleton = {
+  args: {
```
Since the UX design isn't available, I'm not entirely sure what the best approach is for displaying a skeleton for the Tags. However, showing a skeleton would definitely be a good choice.

wbarbee commented:
> > 2. for this one, are you suggesting that we remove these patterns and revisit cards and accordion when those components are migrated over from Foundation?
> 
> No, suggesting maybe making these skeletons actual components instead of sample/patterns. A pre-bundled pattern basically.

understood, but since the card component is deprecated, should we wait to implement vital and informational patterns as a new card component or wait for foundation to be merged with applications?

wbarbee commented:
1) modal, tabs, and side drawer skeleton components removed
2) card pattern skeletons for vital and informational card converted to individual skeleton components within the larger Card component (temporary)
3) refactored skeleton.ts to utilize more generic `shape` and `size` properties instead of `elementType`. Instances of <kyn-skeleton> now rely more heavily on setting width and height, but have some pre-set sizes and shapes. we can add additional sizes, if desired
4) fixed incorrect .inline style selector
5) tagColor has been implemented within the tag skeleton

brian-patrick-3 commented:
:tada: This PR is included in version 1.46.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@kyndryl-design-system/shidoka-applications/v/1.46.0)
- [GitHub release](https://github.com/kyndryl-design-system/shidoka-applications/releases/tag/v1.46.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:

brian-patrick-3 commented:
:tada: This PR is included in version 2.0.0-next.18 :tada:

The release is available on:
- [npm package (@next dist-tag)](https://www.npmjs.com/package/@kyndryl-design-system/shidoka-applications/v/2.0.0-next.18)
- [GitHub release](https://github.com/kyndryl-design-system/shidoka-applications/releases/tag/v2.0.0-next.18)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
